src/Stratis.Bitcoin.Features.SmartContracts/SmartContractFeature.cs

```diff
@@ -42,18 +45,24 @@ public sealed class SmartContractFeature : FullNodeFeature
         private readonly Network network;
         private readonly IStateRepositoryRoot stateRoot;
 
-        public SmartContractFeature(IConsensusManager consensusLoop, ILoggerFactory loggerFactory, Network network, IStateRepositoryRoot stateRoot)
+        public SmartContractFeature(IConsensusManager consensusLoop, ILoggerFactory loggerFactory, Network network, IStateRepositoryRoot stateRoot, ISmartContractActivationProvider smartContractPosActivationProvider = null)
         {
             this.consensusManager = consensusLoop;
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.network = network;
             this.stateRoot = stateRoot;
+
+            if (network.Consensus.ConsensusFactory is SmartContractPoSConsensusFactory)
+            {
+                Guard.NotNull(smartContractPosActivationProvider, nameof(smartContractPosActivationProvider));
```

Comment:
`smartContractPosActivationProvider` isnt set?

Reply:
The 2nd PR mocks this in the tests now.